tests/core/data/test_json_data_node.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+import pathlib
+
+import numpy as np
+import pytest
+
+from src.taipy.core.common.alias import DataNodeId
+from src.taipy.core.data._data_manager import _DataManager
+from src.taipy.core.data.json import JSONDataNode
+from src.taipy.core.exceptions.exceptions import MissingRequiredProperty, NoData
+from taipy.config.config import Config
+from taipy.config.data_node.scope import Scope
+from taipy.config.exceptions.exceptions import InvalidConfigurationId
+
+
+class MyCustomObject:
+    def __init__(self, id, integer, text):
+        self.id = id
+        self.integer = integer
+        self.text = text
+
+
+class MyCustomObject2:
+    def __init__(self, id, boolean, text):
+        self.id = id
+        self.boolean = boolean
+        self.text = text
+
+
+class TestJSONDataNode:
+    def test_create(self):
+        path = "data/node/path"
+        dn = JSONDataNode("foo_bar", Scope.PIPELINE, name="super name", properties={"default_path": path})
+        assert isinstance(dn, JSONDataNode)
+        assert dn.storage_type() == "json"
+        assert dn.config_id == "foo_bar"
+        assert dn.name == "super name"
+        assert dn.scope == Scope.PIPELINE
+        assert dn.id is not None
+        assert dn.parent_id is None
+        assert dn.last_edition_date is None
+        assert dn.job_ids == []
+        assert not dn.is_ready_for_reading
+        assert dn.path == path
+
+        with pytest.raises(InvalidConfigurationId):
+            dn = JSONDataNode(
+                "foo bar", Scope.PIPELINE, name="super name", properties={"default_path": path, "has_header": False}
+            )
+
+    def test_new_csv_data_node_with_existing_file_is_ready_for_reading(self):
+        not_ready_dn_cfg = Config.configure_data_node(
+            "not_ready_data_node_config_id", "json", default_path="NOT_EXISTING.json"
+        )
+        not_ready_dn = _DataManager._bulk_get_or_create([not_ready_dn_cfg])[not_ready_dn_cfg]
+        assert not not_ready_dn.is_ready_for_reading
```

Comment:
We should test that `DataNodeConfig.default_path` correctly populate `datanode.path`:

```suggestion
        not_ready_dn = _DataManager._bulk_get_or_create([not_ready_dn_cfg])[not_ready_dn_cfg]
        assert not not_ready_dn.is_ready_for_reading
        assert not_ready_dn.path == "NOT_EXISTING.json"
```